measurement/plugins/latency/measurements.py

```diff
@@ -0,0 +1,232 @@
+
+        try:
+            minimum_latency = float(match_data.get("minimum_latency"))
+        except (TypeError, ValueError):
+            return [
+                self._get_latency_error(
+                    "ping-minimum-latency", host, traceback=latency_out.stdout
+                )
+            ]
+
+        try:
+            average_latency = float(match_data.get("average_latency"))
+        except (TypeError, ValueError):
+            return [
+                self._get_latency_error(
+                    "ping-average-latency", host, traceback=latency_out.stdout
+                )
+            ]
+
+        try:
+            median_deviation = float(match_data.get("median_deviation"))
+        except (TypeError, ValueError):
+            return [
+                self._get_latency_error(
+                    "ping-median_deviation", host, traceback=latency_out.stdout
+                )
+            ]
+
+        try:
+            latency_data = latency_out.stdout.split("\n")[-3]
+        except IndexError:
+            return [
+                self._get_latency_error(
+                    "ping-split", host, traceback=latency_out.stdout
+                )
+            ]
+
+        matches = LATENCY_PACKETS_REGEX.search(latency_data)
+        try:
+            match_data = matches.groupdict()
+        except AttributeError:
+            return [
+                self._get_latency_error(
+                    "ping-regex", host, traceback=latency_out.stdout
+                )
+            ]
+
+        packets_transmitted = int(match_data.get("packets_transmitted"))
+        packets_received = int(match_data.get("packets_received"))
+        packet_loss = float(match_data.get("packet_loss"))
+        time_taken = float(match_data.get("time"))
+        time_unit = match_data.get("time_unit")
+
+        results = [
+            LatencyMeasurementResult(
+                id=self.id,
+                host=host,
+                minimum_latency=minimum_latency,
+                average_latency=average_latency,
+                maximum_latency=maximum_latency,
+                median_deviation=median_deviation,
+                packets_transmitted=packets_transmitted,
+                packets_received=packets_received,
+                packets_lost=packet_loss,
+                packets_lost_unit=RatioUnit.percentage,
+                time=time_taken,
+                time_unit=TimeUnit(time_unit),
+                errors=[],
+            )
+        ]
+
+        if self.include_individual_results:
+            matches = LATENCY_INDIVIDUAL_PING_REGEX.findall(latency_out.stdout)
+            for match in matches:
+                results.append(
+                    LatencyIndividualMeasurementResult(
+                        id=self.id,
+                        host=host,
+                        errors=[],
+                        packet_size=match[0],
+                        packet_size_unit=StorageUnit(match[1].replace("bytes", "B")),
```

Comment:
The `LATENCY_INDIVIDUAL_PING_REGEX` accept string for `<packet_size_units>` but here's we're assuming `bytes` or something else that'll match the `StorageUnit` enum. If that's not the case will this throw an unhandled exception? Or are we confident enough in the output from `ping` that that isn't a worthwhile consideration?

Reply:
The packet size isn't actually configurable with the current implementation and defaults to `bytes` so I think that is fine for the moment. If we ever want to define the amount it will require revisiting